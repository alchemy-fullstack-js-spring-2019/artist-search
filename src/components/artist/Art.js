import React from 'react';
import PropTypes from 'prop-types';

function Art({ id, archive }) {
  const imgStyles = {
    width: '200px'
  };

  if(archive) {
    return <img style={imgStyles} src={`http://coverartarchive.org/release/${id}/front`} />;
  } else return <img style={imgStyles} src={'http://www.baronblaze.com/wp-content/uploads/2015/12/music-placeholder.png'} />;
}

Art.propTypes = {
  archive: PropTypes.bool.isRequired,
  id: PropTypes.string.isRequired
};

export default Art;
