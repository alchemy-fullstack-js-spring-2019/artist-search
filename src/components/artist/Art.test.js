import React from 'react';
import { shallow } from 'enzyme';
import Art from './Art';

describe('Art component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Art id='' archive={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
