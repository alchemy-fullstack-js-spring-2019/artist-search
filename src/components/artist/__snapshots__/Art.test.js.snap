// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Art component matches snapshot 1`] = `
<img
  src="http://coverartarchive.org/release//front"
  style={
    Object {
      "width": "200px",
    }
  }
/>
`;
